### DISCUSSION TITLE
Glitchy In Certain ProTools Versions

### DISCUSSION THREAD

**Discussion by joshegreenfield2** [iluvcapra/py-ptsl#56] [Q&A]
Hi,

The app I'm building is working great in Protools 2024.10.0 but when I try in a different versions such as 2023.12.0, 2024.6 or 2024.3 the mouse in ProTools starts to glitch and flicker between the ProTools tools (such as hand tool or selector) and the regular Mac mouse. Additionally, it slows down ProTools in these versions. It looks like its directly related to how often the app is checking the Protools tracks names and record armed enabled/disabled status. I have my app set to check these statuses every .5 seconds so that it's essentially live updates, but I can't fgiure out why it works well in one version of PT and not in others.

Thank you,
Josh
PT Ultimate 2024.10.0
Mac Sonoma M2 14.1.1

**Comment by iluvcapra** [comment]
Are you using the synchronous or non-sync api?

**Reply by joshegreenfield2** [reply]
synchronous

**Reply by iluvcapra** [reply]
I think they changed how the server worked in 2024.10, before then it wasn't really designed for continuous polling, it worked on the main event loop so you'd get disturbances working with menus and stuff. Like I used to just have a "refresh" button and then a reload whenever my app was foregrounded, polling wasn't efficient. It's good if they're fixing that.

**Reply by joshegreenfield2** [reply]
Copy that. Thank you for this info. Very helpful!
